Source/VisualStudioTools/Private/VSTestAdapterCommandlet.cpp

```diff
@@ -2,134 +2,153 @@
 
 #include "VSTestAdapterCommandlet.h"
 
-#include <Runtime/Core/Public/Async/TaskGraphInterfaces.h>
-#include <Runtime/Core/Public/Containers/Ticker.h>
-#include <fstream>
+#include "Runtime/Core/Public/Async/TaskGraphInterfaces.h"
+#include "Runtime/Core/Public/Containers/Ticker.h"
 #include <string>
 
 #include "VisualStudioTools.h"
 
-int32 UVSTestAdapterCommandlet::Main(const FString& Params)
+static constexpr auto ListTestsParam = TEXT("listtests");
+static constexpr auto RunTestsParam = TEXT("runtests");
+static constexpr auto TestResultsFileParam = TEXT("testresultfile");
+static constexpr auto HelpParam = TEXT("help");
+
+static void GetAllTests(TArray< FAutomationTestInfo >& OutTestList)
 {
-	TArray<FString> Tokens;
-	TArray<FString> Switches;
-	TMap<FString, FString> ParamVals;
+	FAutomationTestFramework& Framework = FAutomationTestFramework::GetInstance();
+	Framework.GetValidTestNames(OutTestList);
+}
 
-	// Functionality for Unreal Engine Test Adapter.
-	ParseCommandLine(*Params, Tokens, Switches, ParamVals);
-	FAutomationTestFramework::GetInstance().SetRequestedTestFilter( EAutomationTestFlags::PriorityMask | EAutomationTestFlags::ProductFilter | EAutomationTestFlags::SmokeFilter | EAutomationTestFlags::PerfFilter | EAutomationTestFlags::StressFilter | EAutomationTestFlags::NegativeFilter );	
-	if ( ParamVals.Contains( TEXT( "listtests" ) ) )
+static void ReadTestsFromFile(const FString& InFile, TArray< FAutomationTestInfo >& OutTestList)
+{
+	TSet< FString > TestCommands;
+
+	std::wifstream InStream(*InFile);
+	if (!InStream.good())
 	{
-		return ListTests( ParamVals[TEXT( "listtests" )] );
+		UE_LOG(LogVisualStudioTools, Error, TEXT("Failed to open file at path: %s"), *InFile);
+		return;
 	}
-	else if ( ParamVals.Contains( TEXT( "runtests" ) ) && ParamVals.Contains( TEXT( "testresultfile" ) ) )
+
+	std::wstring Line;
+	while (std::getline(InStream, Line))
 	{
-		return RunTests( ParamVals[TEXT( "runtests" )], ParamVals[TEXT( "testresultfile" )] );
+		if (Line.length() > 0)
+		{
+			TestCommands.Add(FString(Line.c_str()));
+		}
 	}
+	InStream.close();
```

Comment:
[minor] 
If `std::wifstream` has RAII capabilities, then I suggest using an inner scope for the object instead of calling `.close()` directly. Would be wrapping lines 26-40 in a pair of `{}`.

Reply:
Will address in future PR soon.